custom_components/pfsense/pypfsense/__init__.py

```diff
@@ -594,6 +594,14 @@ def arp_get_mac_by_ip(self, ip, do_ping=True):
             return None
         return response
 
+    def flush_states(self):
+       
+        script = """
+mwexec("/sbin/pfctl -F state");
```

Comment:
I think here I would do:

```
require_once("filter.inc");
filter_flush_state_table();
```

Reply:
yes I copied the mwexec("/sbin/pfctl -F state"); command from php, but if you dig deep enough, there is a problem with the underlying pfctl arguments. pfsense appears to calling pfctl using a legacy argument, state. I've double checked, on the man page for [pfctl](https://www.freebsd.org/cgi/man.cgi?query=pfctl):

	     -F	nat	   Flush the NAT rules.
	     -F	queue	   Flush the queue rules.
	     -F	rules	   Flush the filter rules.
	     -F	states	   Flush the state table (NAT and filter).
	     -F	Sources	   Flush the source tracking table.
	     -F	info	   Flush the filter information	(statistics that are
			   not bound to	rules).
	     -F	Tables	   Flush the tables.
	     -F	osfp	   Flush the passive operating system fingerprints.
	     -F	all	   Flush all of	the above.

and the docs match the code [pfctl.c](https://github.com/freebsd/freebsd-src/blob/main/sbin/pfctl/pfctl.c)

"- F state" isn't defined, which is curious to me. Abstracting away the underlying operating system version is normally a good rationale for using pfsense's filter_flush_state_table() php wrapper , but then again, pfsense's wrapper is out of date 

I assume "state" is a moniker for "states"? Do you want to submit a pull request to update the pfsense function call?  maybe rename the pfsense method to filter_flush_states_table (or is it filter_flush_rules_table)?

Reply:
okay! confirmed, pfsense is not out of date, there is a typo in the pfctl -F argument call. "state" is clearly not defined going back several years, and pfctl -F stat per forms the same function as pfctl -F states. 

I'll submit a pull request to the pfsense project

Reply:
so I copied the mwexec("/sbin/pfctl -F state"); command from [filter.inc](https://github.com/pfsense/pfsense/blob/master/src/etc/inc/filter.inc), but if you dig deep enough, there is a problem with pfsense pfctl argument usage. pfsense uses the -F arguement state. I've double checked, on the man page for [pfctl](https://www.freebsd.org/cgi/man.cgi?query=pfctl):

	     -F	nat	   Flush the NAT rules.
	     -F	queue	   Flush the queue rules.
	     -F	rules	   Flush the filter rules.
	     -F	states	   Flush the state table (NAT and filter).
	     -F	Sources	   Flush the source tracking table.
	     -F	info	   Flush the filter information	(statistics that are
			   not bound to	rules).
	     -F	Tables	   Flush the tables.
	     -F	osfp	   Flush the passive operating system fingerprints.
	     -F	all	   Flush all of	the above.

and the docs match the code [pfctl.c](https://github.com/freebsd/freebsd-src/blob/main/sbin/pfctl/pfctl.c)

"- F state" isn't defined, which is curious to me. Abstracting away the underlying operating system version is normally a good rationale for using pfsense's filter_flush_state_table() php wrapper , but then again, pfsense's underlying pfctl call is incorrect

Reply:
I submitted [issue 12757](https://redmine.pfsense.org/issues/12757) to the pfsense project